playbooks/roles/mattermail/defaults/main.yml

```diff
@@ -0,0 +1,18 @@
+# Connections settings for Mattermost
+MATTERMOST_URL: https://mattermost.example.com
+MATTERMOST_TEAM: opencraft
+MATTERMOST_LOGIN: mattermail
+MATTERMOST_PASSWORD: set-me-please
+MATTERMOST_USER: set-me-please
+MATTERMOST_PASS: set-me-please
+MATTERMOST_CHANNEL: hosting-subteam
+
+# Mattermail settings to fetch emails
+MATTERMAIL_SERVER: imap.example.com:143
+MATTERMAIL_USERNAME: urgent
+MATTERMAIL_PASSWORD: set-me-please
+
+# Settings that can be left at their default values
+MATTERMAIL_REPO: github.com/rodcorsi/mattermail
```

Comment:
Or if they have release tags set up you can set that up, and we can update the tag when there is a new release?

Reply:
Unfortunately the `go get` command used to fetch the repository and build the binary only supports fetching the `master`

Reply:
I looked at this again, and just remembered that since it's a Go package, it will 100% already upload the binary for easy download, if it's well maintained that is. And of course, it does: https://github.com/rodcorsi/mattermail/releases/download/3.5.0/mattermail-3.5.0.linux.am64.tar.gz

So we should add this as a variable URL, and insert the version as another variable as necessary. Then we can use `unarchive`, similar to [this example](https://github.com/open-craft/ansible-playbooks/blob/master/playbooks/roles/consul/tasks/main.yml#L33-L40).

But then again, about security, I don't know enough about Go to know if these binaries are somehow sandboxed or something -- I'm guessing not? (Even then we can't be sure we're getting a Go binary). If it's just a matter of trust (e.g., we download the binary for Consul because we trust Hashicorp wouldn't give us something malicious), then I guess we can't assume trust for Mattermail, and would probably need our own repo...

@xitij2000 @viadanna what do you guys think? cc @bradenmacdonald

Reply:
@UmanShahzad  Code isn't any safer than a binary if it hasn't been audited :-) Do we fully 100% trust the code to begin with? Because the code and binary are from the same source.